src/Orleans/CodeGeneration/KnownTypeAttribute.cs

```diff
@@ -2,17 +2,34 @@ namespace Orleans.CodeGeneration
 {
     using System;
 
-    /// <summary>
-    /// The attribute which informs the code generator that code should be generated a type.
-    /// </summary>
-    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
-    public class KnownTypeAttribute : Attribute
+    public class ConsiderForCodeGenerationAttribute : Attribute
     {
-        public KnownTypeAttribute(Type type)
+        protected ConsiderForCodeGenerationAttribute(Type type, bool generateSerializer)
         {
             this.Type = type;
+            this.GenerateSerializer = generateSerializer;
         }
 
-        public Type Type { get; set; }
+        public Type Type { get; private set; }
+        public bool GenerateSerializer { get; private set; }
+    }
+
+    /// <summary>
+    /// The attribute which informs the code generator that code should be generated for this type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public class KnownTypeAttribute : ConsiderForCodeGenerationAttribute
+    {
+        public KnownTypeAttribute(Type type) : base(type, false){}
+    }
+
+    /// <summary>
+    /// The attribute which informs the code generator that code should be generated for this type.
+    /// Forces generation of type serilizer.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public class GenerateSerializerAttribute : ConsiderForCodeGenerationAttribute
```

Comment:
please mark all concrete attributes as `sealed` (according to [CA1813](https://msdn.microsoft.com/en-us/library/ms182267(v=vs.140).aspx))